### PR TITLE
📖 ClusterClass: restructure documentation

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
First step of #5953

### CODE REVIEW

**Review by fabriziopandini — COMMENTED** [review]
overall lgtm, only tiny nits

File: docs/book/src/SUMMARY.md [hunk 1]
```diff
@@ -20,8 +20,8 @@
         - [ClusterResourceSet](./tasks/experimental-features/cluster-resource-set.md)
         - [ClusterClass](./tasks/experimental-features/cluster-class/index.md)
             - [Writing a ClusterClass](./tasks/experimental-features/cluster-class/write-clusterclass.md)
-            - [Upgrading a Cluster](./tasks/experimental-features/cluster-class/upgrade-cluster.md)
             - [Changing a ClusterClass](./tasks/experimental-features/cluster-class/change-clusterclass.md)
+            - [Operating a Cluster](./tasks/experimental-features/cluster-class/operate-cluster.md)
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply docs/book/src/SUMMARY.md]: Sounds good, updated

**Pull request merged**